### PR TITLE
Consistently use try_ prefix on fallible functions

### LINKED ISSUE #2576: Rename all fallible from_bytes functions to try_from_bytes
Many, or most, of our functions that are named `from_bytes` are fallible. @markusicu suggested that we follow the Rust convention of `try_from_...` to indicate a fallible function.

Thoughts?

- [ ] @zbraniecki 
- [x] @Manishearth

### PR DESCRIPTION
**Pull request by sffc**
Fixes #2576

### CODE REVIEW

File: components/calendar/src/any_calendar.rs [hunk 1]
```diff
@@ -608,14 +607,14 @@ impl AnyCalendarKind {
     /// Construct from a BCP-47 string
     ///
     /// Returns None if the calendar is unknown
-    pub fn from_bcp47_string(x: &str) -> Option<Self> {
-        Self::from_bcp47_bytes(x.as_bytes())
+    pub fn try_from_bcp47_string(x: &str) -> Result<Self, DateTimeError> {
```
Comment (Manishearth) [inline components/calendar/src/any_calendar.rs:610]: issue: I was actually hoping to go the other way here. Part of the problem here is that `DateTimeError::unknown_kind_from_bytes()` allocates, and if you're immediately throwing away the error it's wasted work. I'd prefer if these all uniformly returned `Option`, and callers did `.ok_or_else(|| DateTimeError::unknown_kind_from_bytes(s))` when necessary.

Comment (Manishearth) [inline components/calendar/src/any_calendar.rs:610]: issue: In our discussion I  was actually suggesting we go the other way here. Part of the problem here is that `DateTimeError::unknown_kind_from_bytes()` allocates, and if you're immediately throwing away the error it's wasted work. I'd prefer if these all uniformly returned `Option`, and callers did `.ok_or_else(|| DateTimeError::unknown_kind_from_bytes(s))` when necessary.

**Review by Manishearth — APPROVED** [review]

File: components/calendar/src/any_calendar.rs (hunk 1 above)
Comment (sffc) [reply components/calendar/src/any_calendar.rs:610]: It only allocates in the `Value` version of the function.

But I see your point about code size.

Comment (sffc) [reply components/calendar/src/any_calendar.rs:610]: How about `AnyCalendarKind::get_for_bcp47_string -> Option` ?

File: ffi/diplomat/src/timezone.rs [hunk 2]
```diff
@@ -108,7 +108,7 @@ pub mod ffi {
             self.0
                 .gmt_offset
                 .map(|v| v.is_positive())
-                .ok_or(ICU4XError::TimeZoneMissingInputError)
+                .ok_or(ICU4XError::OptionNoneError)
```
Comment (sffc) [inline ffi/diplomat/src/timezone.rs:111]: Return `()` instead of `OptionNoneError`

**Review by Manishearth — APPROVED** [review]

File: ffi/diplomat/src/errors.rs [hunk 3]
```diff
@@ -47,9 +47,10 @@ pub mod ffi {
         /// Typically found when not enough space is allocated
         /// Most APIs that return a string may return this error
         WriteableError = 0x01,
-
         // Some input was out of bounds
         OutOfBoundsError = 0x02,
+        // An optional value is not present (not necesarilly an error)
```
[the remark this answers is not in the document]
Comment (sffc) [reply ffi/diplomat/src/errors.rs]: (OptionNoneError is no longer a thing)

**Pull request merged**